test/viewmodels/DutyDefermentAccountViewModelSpec.scala

```diff
@@ -0,0 +1,247 @@
+/*
+ * Copyright 2023 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package viewmodels
+
+import config.AppConfig
+import org.scalatest.Assertion
+import play.api.Application
+import play.api.i18n.Messages
+import play.twirl.api.Html
+import uk.gov.hmrc.hmrcfrontend.views.html.components.HmrcNewTabLink
+import uk.gov.hmrc.hmrcfrontend.views.viewmodels.newtablink.NewTabLink
+import util.SpecBase
+import views.html.components.{caption, h1, h2, inset, link, p}
+import views.html.requested_statements
+
+class DutyDefermentAccountViewModelSpec extends SpecBase {
+
+  "apply method" must {
+
+    "populate the model correctly" when {
+
+      "current statements are available" in new Setup {
+        implicit val appConfig: AppConfig = app.injector.instanceOf[AppConfig]
```

Comment:
can `appConfig` and `msgs` be in the trait object

Reply:
Moved to the trait and reusing these